tools/lib/bpf/relo_core.c

```diff
@@ -1213,11 +1188,11 @@ int bpf_core_apply_relo_insn(const char *prog_name, struct bpf_insn *insn,
 
 	/* TYPE_ID_LOCAL relo is special and doesn't need candidate search */
 	if (relo->kind == BPF_CORE_TYPE_ID_LOCAL) {
-		targ_res.validate = true;
-		targ_res.poison = false;
-		targ_res.orig_val = local_spec->root_type_id;
-		targ_res.new_val = local_spec->root_type_id;
-		goto patch_insn;
+		targ_res->validate = true;
+		targ_res->poison = false;
+		targ_res->orig_val = local_spec->root_type_id;
+		targ_res->new_val = local_spec->root_type_id;
+		return 0;
```

Comment:
Is `patch_insn` label called after the invocation of `bpf_core_calc_relo_insn` in any case?

Reply:
Following the code flow, it seems it is, worth a double-check...

Reply:
Yes, it's called. The return value is 0 and `bpf_object__relocate_core` ends up calling `bpf_core_patch_insn`.

Reply:
Yep, I tracked that flow. Not sure if there were others I couldn't spot :)